src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,38 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
-import core.basesyntax.model.User;
+import core.basesyntax.service.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_AGE = 18;
+    private static final int MIN_LOGIN_AND_PASSWORD_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
-    public User register(User user) {
-        return null;
+    public User register(User user) throws RegistrationException {
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RegistrationException("Login: " + user.getLogin()
+                    + " is already taken! Choose another one");
+        }
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login can't be null");
+        }
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password can't be null");
+        }
+        if (user.getAge() == null) {
+            throw new RegistrationException("Age can't be null");
+        }
+        if (user.getLogin().length() < MIN_LOGIN_AND_PASSWORD_LENGTH) {
+            throw new RegistrationException("Login length must be at least 6 characters");
+        }
+        if (user.getPassword().length() < MIN_LOGIN_AND_PASSWORD_LENGTH) {
+            throw new RegistrationException("Password length must be at least 6 characters");
+        }
+        if (user.getAge() < MIN_AGE) {
+            throw new RegistrationException("Not valid age: " + user.getAge()
+                    + ". Min allowed age is " + MIN_AGE);
+        }
+        return storageDao.add(user);
```

Comment:
This method can potentially return null if the 'storageDao.add(user)' method returns null. Ensure that the 'add' method in 'StorageDaoImpl' is implemented correctly to return the added user.